husky_components/collection-navigation/main.js

```diff
@@ -0,0 +1,342 @@
+     * @event husky.collection-navigation.collection.change
+     */
+    CHANGE_COLLECTION = createEventName('collection.change'),
+
+    /**
+     * raised when clicked on the add collection button
+     * @event husky.collection-navigation.collection.add
+     */
+    ADD_COLLECTION = createEventName('collection.add'),
+
+    /**
+     * raised when clicked on the item icon
+     * @event husky.collection-navigation.content.show
+     */
+    SHOW_CONTENT = createEventName('content.show');
+
+    return {
+        /**
+         * @method initialize
+         */
+        initialize: function() {
+            this.cache = null;
+            this.collectionView = null;
+            this.options = this.sandbox.util.extend(true, {}, defaultOptions, this.options);
+            this.template = this.sandbox.util.template(mainTpl);
+            this.headerTpl = this.sandbox.util.template(headerTpl);
+            this.render();
+
+            return this.load().then(function(data) {
+                this.sandbox.emit(INITIALIZED);
+
+                this.collectionView = this.createCollectionView(data);
+                this.updateHeader(data);
+                this.storeData(data);
+                this.appendCollectionView();
+            }.bind(this));
+        },
+
+        /**
+         * Component destructor function
+         * @method remove
+         */
+        remove: function() {
+            this.cache.destroy();
+            this.collectionsList.destroy();
+        },
+
+        /**
+         * Initial render method. Inserts the main template into the dom
+         * @method render
+         */
+        render: function() {
+            var tpl = this.template();
+            this.$el.html(tpl);
+            this.bindDOMEvents();
+        },
+
+        /**
+         * @method bindDOMEvents
+         */
+        bindDOMEvents: function() {
+            this.$el.on('click', '.collection-navigation-item-name', this.openChildCollectionHandler.bind(this));
+            this.$el.on('click', '.collection-navigation-item-thumb', this.showContentHandler.bind(this));
+            this.$el.on('click', '.collection-navigation-back', this.openParentCollectionHandler.bind(this));
+            this.$el.on('click', '.collection-navigation-add', this.addCollectionHandler.bind(this));
+        },
+
+        /**
+         * Fetch the data from the server
+         * @method load
+         * @param {String} url
+         */
+        load: function(url) {
+            url = url || this.options.url;
+
+            return this.sandbox.util.load(url)
+                        .then(this.parse.bind(this));
+        },
+
+        /**
+         * @method parse
+         * @param {Object} response Response
+         */
+        parse: function(response) {
+            var current = {},
+                parent = response._embedded.parent;
+
+            current.id = response.id || consts.ROOT_ID;
+            current.name = response.name;
```

Comment:
in the root there is no name please provide a default value which is translated (can be set over options).

Additionally change name to a configurable property name (default title)